shared/src/commonMain/kotlin/org/hyperskill/app/analytic/domain/interactor/AnalyticInteractor.kt

```diff
@@ -43,58 +27,34 @@ class AnalyticInteractor(
     }
 
     override suspend fun flushEvents() {
-        launchHyperskillFlushEventsJob(withDelay = false)
+        hyperskillAnalyticEngine.flushEvents()
+        appsFlyerAnalyticEngine?.flushEvents()
+
         eventMonitor?.analyticDidFlushEvents()
     }
 
     suspend fun logEvent(event: AnalyticEvent, forceLogEvent: Boolean = false) {
         when (event.source) {
-            AnalyticSource.HYPERSKILL_API -> logHyperskillEvent(event, forceLogEvent)
+            AnalyticSource.HYPERSKILL_API -> {
+                hyperskillAnalyticEngine.reportEvent(event, force = forceLogEvent)
+            }
+            AnalyticSource.APPS_FLYER -> {
+                appsFlyerAnalyticEngine?.reportEvent(event, force = forceLogEvent)
+            }
```

Comment:
I suggest replace `when` expression with  something like this
```kotlin
analyticEngines
            .filter { it.targetSource == event.source }
            .forEach { it.reportEvent(event, forceLogEvent) }
```
In my opinion, it's better to not expose the engines internals to the Analytics interactor.